backend/src/main/java/gov/cdc/usds/simplereport/db/model/DataHubUpload.java

```diff
@@ -0,0 +1,149 @@
+package gov.cdc.usds.simplereport.db.model;
+
+import gov.cdc.usds.simplereport.config.simplereport.DataHubConfig;
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.annotations.DynamicUpdate;
+import org.hibernate.annotations.Type;
+import org.hibernate.annotations.UpdateTimestamp;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.boot.context.properties.ConstructorBinding;
+import org.springframework.data.annotation.LastModifiedDate;
+import org.springframework.data.annotation.ReadOnlyProperty;
+
+import javax.persistence.*;
+import java.time.Instant;
+import java.util.Date;
+import java.util.UUID;
+
+@Entity
+@DynamicUpdate
+public class DataHubUpload {
+
+    @Transient
+    public static final String SUCCESS_JOB = "SUCCESS";
+
+    protected DataHubUpload() { /* no-op for hibernate */ }
+
+    // We have to pass config into the constructor because Configuration doesn't work in constructors
+    @ConstructorBinding
+    public DataHubUpload(@NotNull DataHubConfig config) {
+        createdBy = UUID.fromString(config.getServiceUuid());
+        updatedBy = createdBy;
+        jobState = "INIT";
+        recordsProcessed = 0;
+        responseData = "{}";
+        // give some non-null default so things (like logging) don't fail.
+        earliestRecordedTimestamp = Date.from(Instant.parse("2020-01-01T00:00:00Z"));
+        latestRecordedTimestamp = Date.from(Instant.parse("2020-01-01T00:00:00Z"));
+    }
+
+    @Column(updatable = false, nullable = false)
+    @Id
+    @GeneratedValue(generator = "UUID4")
+    private UUID internalId;
+
+    // set to "SUCCESS" when done.
+    @Column(nullable = false)
+    private String jobState;
+
+    @Column(updatable = false)
+    @Temporal(TemporalType.TIMESTAMP)
+    @CreationTimestamp
+    private Date createdAt;
+
+    @Column(updatable = false)
+    @ReadOnlyProperty
+    private UUID createdBy;
+
+    @Column
+    @Temporal(TemporalType.TIMESTAMP)
+    @UpdateTimestamp
```

Comment:
This gets handled by Hibernate, right? It's a little weird to have JPA interceptors doing one thing and Hibernate interceptors doing another, but I think that ship has sailed.

Possibly we should have this and ApiUser have some shared base code?